sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/BaseLoadBalancerInterceptor.java

```diff
@@ -50,28 +57,47 @@ public class BaseLoadBalancerInterceptor extends AbstractInterceptor {
 
     private final boolean canLoadZuul;
 
+    private final RouterConfig routerConfig;
+
     /**
      * Constructor
      */
     public BaseLoadBalancerInterceptor() {
         loadBalancerService = PluginServiceManager.getPluginService(LoadBalancerService.class);
         canLoadZuul = canLoadZuul();
+        routerConfig = PluginConfigManager.getPluginConfig(RouterConfig.class);
     }
 
     @Override
     public ExecuteContext before(ExecuteContext context) {
         Object object = context.getObject();
-        if (object instanceof BaseLoadBalancer) {
-            List<Object> serverList = getServerList(context.getMethod().getName(), object);
-            if (CollectionUtils.isEmpty(serverList)) {
-                return context;
-            }
-            BaseLoadBalancer loadBalancer = (BaseLoadBalancer) object;
-            String name = loadBalancer.getName();
-            RequestData requestData = getRequestData().orElse(null);
-            List<Object> targetInstances = loadBalancerService.getTargetInstances(name, serverList, requestData);
-            context.skip(Collections.unmodifiableList(targetInstances));
+        if (!(object instanceof BaseLoadBalancer)) {
+            return context;
+        }
+        BaseLoadBalancer loadBalancer = (BaseLoadBalancer) object;
+        String name = loadBalancer.getName();
+        RequestData requestData = getRequestData().orElse(null);
+
+        // use xds route to find service instances
+        Set<ServiceInstance> serviceInstanceByXdsRoute = null;
+        if (routerConfig.isEnabledXdsRoute()) {
+            serviceInstanceByXdsRoute = XdsRouterHandler.INSTANCE
+                    .getServiceInstanceByXdsRoute(name, requestData.getPath(),
+                            BaseHttpRouterUtils.processHeaders(requestData.getTag()));
+        }
+        if (serviceInstanceByXdsRoute != null && !serviceInstanceByXdsRoute.isEmpty()) {
+            context.skip(Collections.unmodifiableList(SpringRouterUtils
+                    .getSpringCloudServiceInstanceByXds(serviceInstanceByXdsRoute)));
+            return context;
         }
```

Comment:
xds logic can be extracted as a function

Reply:
done